### DISCUSSION TITLE
Nowy design serwisu

### DISCUSSION THREAD

**Discussion by TomaszGasior** [TomaszGasior/RadioLista-v3#139] [2. Pomysły Ideas]
Pracuję nad nowym designem RadioListy. Aktualny design pochodzi z 2016 roku i zestarzał się: ogranicza moje możliwości i źle prezentuje się na urządzeniach mobilnych. Zadanie wprowadzenia nowego designu od wielu miesięcy blokuje inne zadania (takie jak #120  i #121), tak więc dopóki go nie skończę, trudno mi będzie wprowadzać nowe funkcje. Nic nie mogę obiecać ze względu na zajęty czas, ale chciałbym go dokończyć w ciągu najbliższych miesięcy.

Cele nowego designu #116  są następujące:
- nowe logo: prawie takie samo,  ale mniej krzywe;
- nawigację opartą na kartach  (np. w ustawieniach wykazu, ekranie dodawania stacji) zastępuję układem dwukolumnowym: w lewej kolumnie wąskie menu, w prawej treść po prostu przewijana góra-dół (wstępna wersja: 
[Nagranie ekranu z 05.05.2021 22:05:44.zip](https://github.com/TomaszGasior/RadioLista-v3/files/6429987/Nagranie.ekranu.z.05.05.2021.22.05.44.zip));
- bardziej eleganckie kolory;
- niektóre bardzo opasłe sekcje ustawień, np. kolumny wykazu, wyciągnę do oddzielnych podstron;
- chciałbym poprawić trochę dostępność (nie tylko dla niepełnosprawnych): np. kontrast kolorów; zachowanie formularzy przy błędnie wypełnionych polach;
- chciałbym też wprowadzić opcjonalny dark mode, czyli powszechny w wielu serwisach tryb ciemny.

Gdy nowy design będzie gotowy, udostępnię wam go miesiąc przed wrzuceniem na radiolista.pl na domenę testową dev.radiolista.pl, gdzie będzie można sobie go przetestować i ew. wyrazić opinię. Nie będzie diametralnych zmian: jedynie drobne poprzekładanie rzeczy tu i tam, żeby łatwiej było wprowadzać nowe funkcje. 

**Jeżeli ktoś jest zainteresowany przetestowaniem nowego designu, gdy będzie gotowy, proszę klikać „Subscribe” po prawej stronie ekranu. Bedę pisać.**

**Comment by HyperDX** [comment]
Ciekawie się to prezentuje, trzymam kciuki i czekam na efekty :).

**Comment by TomaszGasior** [comment]
Odkopuję temat. :) Nowy design zostanie wdrożony w 10 urodziny RadioListy — 1 lipca. Na początku czerwca udostępnię wam dostęp po podglądu nowego designu pod oddzielnym adresem, abyście mogli go przetestować i wyrazić ewentualne uwagi. Długo to zajęło, bo temat ten miałem zamknąć już w 2020, ale czasem tak bywa. :)

**Comment by TomaszGasior** [comment]
Udało mi się zrealizować plan minimum związany z nowym designem serwisu. Wciąż nie wszystko jest gotowe i nie wiem, które dodatkowe rzeczy być może zdążę jeszcze zrobić przed wdrożeniem nowej szaty graficznej 1 lipca. Tematy, które jeszcze są niedokończone: #121 #178 #187 #203 #202 

Zgodnie z obietnicą daję wam przedpremierowy dostęp do nowej wersji serwisu. Będe wdzięczny za przetestowanie i wyrażenie opinii. Co się zmieniło, co poprawiłem:
- bardziej nowoczesna szata graficzna,
- większa ergonomia: wszystko bardziej na wyciągnięcie ręki,
- łatwiejsze usuwanie stacji,
- tu i ówdzie poprawiony kontrast,
- poprawki przejrzystości formularzy, np. komunikatów błędów i uwag pomocy.

<s>**Testowa wersja jest dostępna pod adresem [usunięte].** Serwis jest wypełniony losowo wygenerowanymi danymi przykładowymi, które mogą czasem nie mieć sensu. </s>

**Reply by HyperDX** [reply]
Chwilkę przetestowałem i jestem pod wrażeniem zmian. O wiele przyjemniej mi się pracuje przy dodawaniu stacji w takim układzie niż w starej RL z przeklikiwaniem każdej z kart. Bardzo ładny layout CSS ogólnie świetnie się to prezentuje pod kątem wizualnym. Usuwanie wielu stacji działa i może być bardzo przydatne dla niektórych. Dobra robota!

**Comment by TomaszGasior** [comment]
Obecnie strona główna instancji testowej wyrzuca błąd 502, podobnie jak kiedyś główna wersja aplikacji, co było zgłaszane w innym wątku: #176 Udało mi się potwierdzić problem i mam nadzieję rozwiązać usterkę #212 

Aby ominąć usterkę, wystarczy dopisać `/logowanie` albo `/moje-wykazy` na końcu w pasku adresu przeglądarki i nie wchodzić na stronę główną.

**Comment by TomaszGasior** [comment]
Nowy design serwisu zostaje właśnie wdrożony. Niestety nie udało się ukończyć wszystkich tematów (https://github.com/TomaszGasior/RadioLista-v3/issues/121 https://github.com/TomaszGasior/RadioLista-v3/issues/187  https://github.com/TomaszGasior/RadioLista-v3/issues/203 https://github.com/TomaszGasior/RadioLista-v3/issues/202), niemniej mam nadzieję, że nowy wygląd przypadnie wam do gustu i będzie się wam wygodniej używało serwisu.
